app/controllers/users_controller.rb

```diff
@@ -1,4 +1,12 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
```

Comment:
thiếu show rồi, show cũng phải check logged chứ nè. Chuyển qua dùng excpet cho code gọn hơn nha
```suggestion
  before_action :logged_in_user, except: %i(new create)
```